README.md

```diff
@@ -317,6 +317,73 @@ print(response) # Okay, let me get the current weather report for Amsterdam usin
+
+1. **Strong Type Validation**: Pydantic enforces strict type checking and validation at runtime
+2. **Clear Interface Documentation**: Input/output schemas are automatically generated from your models. The LLM "reads" both the model's docstring and the `description` attributes of Pydantic Field objects to understand how to use the tool correctly. This natural language documentation helps the LLM make informed decisions about parameter values.
+3. **Error Handling with Self-Correction**: Built-in error handling and validation messages are fed back to the LLM, allowing it to understand what went wrong and self-correct its tool usage. For example, if the LLM provides an invalid value for a parameter, Pydantic's detailed error message helps the LLM understand why it was invalid and how to fix it in subsequent attempts.
+4. **Fine-grained Input Control**: Pydantic Field attributes like `pattern` allow you to enforce very specific requirements through regex patterns. For example, you can ensure a parameter only accepts certain string formats, numeric ranges, or specific enumerated values. The LLM understands these constraints from the Field descriptions and validates its choices accordingly.
+5. **Schema Evolution**: Easy to evolve your tool's interface while maintaining backward compatibility
+6. **OpenAPI Integration**: Automatic OpenAPI schema generation for documentation
+
+You can find a complete example in `examples/pydantic_tools/` that demonstrates this approach. The example implements a weather alerts tool with proper input validation, error handling, and response structuring:
+
+```python
+from pydantic import BaseModel, Field
+
+class WeatherAlertRequest(BaseModel):
+    """
+    Request parameters for the weather alerts tool.
+    """
+    area: Optional[str] = Field(
+        default=None,
+        description="State code (e.g., 'CA', 'TX') or zone/county code to filter alerts by area."
+    )
+    severity: Optional[str] = Field(
+        default=None,
+        description="Filter by severity level: 'Extreme', 'Severe', 'Moderate', 'Minor', or 'Unknown'.",
+        pattern="^(Extreme|Severe|Moderate|Minor|Unknown)$"
+    )
+
+class WeatherAlertsTool:
+    @property
+    def tool_spec(self) -> Dict[str, Any]:
+        """Tool specification is automatically generated from the Pydantic model."""
+        schema = WeatherAlertRequest.model_json_schema()
+        return {
+            "name": "get_weather_alerts",
+            "description": WeatherAlertRequest.__doc__,
+            "inputSchema": {"json": schema}
+        }
+
+    def invoke(self, **kwargs) -> Dict[str, Any]:
+        """
+        Invoke the weather alerts tool with validated parameters.
+        """
+        try:
+            request = WeatherAlertRequest(**kwargs)  # Validation happens here
+            return self._get_weather_alerts(request)
+        except ValidationError as e:
+            return {"error": str(e)}
+```
+
+Key advantages of this approach:
```

Comment:
Let's nuke the text below this point as it seems largely redundant with the text above the code sample